examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb

```diff
@@ -143,11 +143,9 @@
      "name": "stderr",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #32.    outputs = outputs &gt;&gt; Dropna()
</pre><p>At this point there  are no nan user IDs because of the coercison to int32 at the top of this cell.</p><p> </p><p>the nan user_ids have already been coerced to -2147483648. So no rows with nan user ids will be dropped as part of this op.</p><pre class="ql-syntax" spellcheck="false">np.array(np.nan, dtype=np.float32).astype(np.int32)
# =&gt; array(-2147483648, dtype=int32)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Merlin/pull/629/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
My example was not representative of what is actually happening in the dataset we're testing with. and how it flows through NVTabular.

cudf by default will preserve both None and nan values as <NA> unlike numpy and pandas

```python
import cudf
import numpy as np

cudf.Series(["1", None]).astype("int32")
# =>
# 0       1
# 1    <NA>
# dtype: int32

cudf.Series([1, np.nan]).astype("int32")
# =>
# 0       1
# 1    <NA>
# dtype: int32

cudf.Series([1.0, np.nan], nan_as_null=False).astype("int32")
# 0             1
# 1   -2147483648
# dtype: int32
```

In this case the Dropna() should achieve what we want here because the NA values will propagate through the whole graph.